### ISSUE TITLE
Implement a "FieldCompoundTile" and a "BehaviorTile"

### ISSUE THREAD

**Issue by rnixx** [plone/plone.app.mosaic#127]
While working on a p.a.mosaic integration with @agitator we encountered the need for more complex tiles.

Usecases:

* Edit date range data (plone.app.event)
* Display summary event details (plone.app.event)
* Edit some control data for standard fields (like scale for a lead image)

The related fields should show up in the tile edit form and the fields should be available in the data attribute of the tile for displaying.

The concrete field compound tile may be a subclass with an attribute compound_fields or similar defining which fields get included, e.g.:

```
class MyCompoundTile(FieldCompoundTile):
    compound_fields = [
        'title',
        'IEventBasic.start',
        'IEventBasic.end',
        'IEventBasic.whole_day',
        'IEventRecurrence.recurrence'
    ]
```

If given entry is dot seperated, i points to a behavior.
If not, it's a field name of the base type.

Note: When adopting latest plone.behavior changes to existing standard behaviors (PLIP 20239, https://github.com/plone/plone.behavior/pull/2 ), the behavior identification can be done with the name instead of the interface (), i.e. 'event_basic.start'.

Respective, if a BehaviorTile is desired as well, a concrete class may look like:

```
class MyBehaviorTile(BehaviorTile):
    behavior = 'event_basic'
```

or via interface...

```
@adapter(IEventBasic)
class MyBehavior(BehaviorTile):
    pass
```

No idea whether it can be implemented using adapter decorator though.

When implementing all this there should also be the possibility to reduce single field type choice and another category should be added for compound/behavior tiles.

@datakurre @pbauer 

What are you thinking about this?

**Comment by datakurre** [comment]
Currently, when a field tile is placed, it should be removed from properties form (if it works properly). What should happen with behaviortile?

**Comment by agitator** [comment]
yesss!

**Comment by datakurre** [comment]
Also, currenty there is only one field tile in code and individual field tiles available in Insert menu are just that same tile with different configuration. Any opinions on that?

**Comment by rnixx** [comment]
@datakurre When implementing all this there should also be the possibility to reduce single field type choice and another category should be added for compound/behavior tiles.

**Comment by agitator** [comment]
I would disable fields (from the fields column) that are used somewhere else. 
If one decides to use a field twice, it should still be possible within another compound tile.

**Comment by datakurre** [comment]
Must also note, that it's already possible to build any kind of tiles (also reading regular fields), but those won't necessarily work nice with field tiles and cannot store to regular (content type) fields, because saving the edit form would override those changes.

**Comment by rnixx** [comment]
@datakurre yes, custom tiles for displaying multiple fields is not really the issue here, it's more editing of multiple fields as one tile

**Comment by thet** [comment]
@rnixx @datakurre this sounds to me like a lot like related to collective.instancebehavior - maybe not code wise, but use case wise. having the possibility to add behaviors tiles to a content instance allows to have a super simple basic content type with no fields, where you add behaviors like title, description, text, event dates, event recurrence and so on.

@rnixx @agitator did you experiment with your idea a bit more?

**Comment by datakurre** [comment]
@thet Not yet. This would be required for both normal and instance behavior. It's not about activating behaviors, but being able to edit them as whole (because fields are often interdependent), not just one field at time.

That said, instance behaviors (and activating them in fly from Mosaic editor) might make sense in the possible world of single Page content type :)

**Comment by rnixx** [comment]
@datakurre: @thet, @jensens and me discussed collective.instancebehavior and thought it would be very nice if it's possible to apply behaviors per instance in general. This also has been mentioned in the discussion about Dexterity API on the mailinglist some time ago. But this needs a PLIP and all the surrounding.

Anyway, once a "BehaviorTile" and generic Instance Behaviors are implemented it's not very far having the kind of "Lego" type and view creation thet mentiones.
